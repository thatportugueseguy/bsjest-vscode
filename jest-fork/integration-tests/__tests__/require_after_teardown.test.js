/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 */

'use strict';

const runJest = require('../runJest');

test('prints useful error for requires after test is done', () => {
  const {stderr} = runJest('require-after-teardown');

  const interestingLines = stderr
    .split('\n')
    .slice(9, 18)
    .join('\n');

  expect(interestingLines).toMatchSnapshot();
  expect(stderr.split('\n')[19]).toMatch(
    new RegExp('(__tests__/late-require.test.js:11:20)'),
  );
});
