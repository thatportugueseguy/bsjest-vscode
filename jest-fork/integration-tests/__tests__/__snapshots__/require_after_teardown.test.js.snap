// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prints useful error for requires after test is done 1`] = `
"ReferenceError: You are trying to \`import\` a file after the Jest environment has been torn down.

       9 | test('require after done', () => {
      10 |   setTimeout(() => {
    > 11 |     const double = require('../');
         |                    ^
      12 | 
      13 |     expect(double(5)).toBe(10);
      14 |   }, 0);"
`;
